src/modules/ibc.ts

```diff
@@ -155,14 +164,116 @@ export class IBCModule extends BaseModule {
+      } 
+    }
+    const chainInfoJson = await chainInfoResponse.json();
+    return chainInfoJson as ChainInfo;
+  }
+
+  async getAssembledChainInfo(chainsData: ChainRegistryItem[], chainId: string, bridgeChainName: string): Promise<ChainInfo | undefined> {
+    const selectedChainData = chainsData.find((chainData: ChainRegistryItem) => {
+      return chainData.chain_name.includes(bridgeChainName.toLowerCase()) || chainData.chain_id.includes(bridgeChainName.toLowerCase());
+    });
+
+    if (selectedChainData) {
+      try {
+        const chainInfoResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/chain.json`);
+        const chainInfoJson = await chainInfoResponse.json();
+        const chainAssetListResponse = await fetch(`https://raw.githubusercontent.com/cosmos/chain-registry/master/${selectedChainData.chain_name}/assetlist.json`);
+        const assetListJson = await chainAssetListResponse.json();
+        const features: string[] = [];
+      if (selectedChainData.cosmwasm_enabled) {
+        features.push("cosmwasm");
+      }
+
+      // Extract the staking currency denomination from chainInfoJson (min. denom)
+      const stakingCurrencyDenom = chainInfoJson['staking']['staking_tokens'][0]['denom'];
+      
+      // Find the asset that matches the stakingCurrencyDenom
+      const stakingAsset = assetListJson.assets.find((asset: Asset) => {
+        return asset.denom_units.some((unit: any) => unit.denom === stakingCurrencyDenom);
+      });
+
+      // Find the highest denom unit
+      const maxExponentDenomUnit = stakingAsset.denom_units.reduce((prev: DenomUnit, current: DenomUnit) => {
+        return (prev.exponent > current.exponent) ? prev : current;
+      });
+
+      let stakeCurrency: Currency = {
+        coinDenom: maxExponentDenomUnit.denom.toUpperCase(),
+        coinMinimalDenom: stakingCurrencyDenom,
+        coinDecimals: maxExponentDenomUnit.exponent,
+        coinGeckoId: stakingAsset.coingecko_id,
+      };
+
+
+      const currencies = assetListJson.assets.map((asset: Asset) => {
+        // Find the denom_unit with the maximum exponent
+        const maxExponentDenomUnit = asset.denom_units.reduce((prev, current) => {
+          return (prev.exponent > current.exponent) ? prev : current;
+        });
+        return {
+          coinDenom: maxExponentDenomUnit.denom.toUpperCase(),
+          coinMinimalDenom: asset.base,
+          coinDecimals: maxExponentDenomUnit.exponent,
+          coinGeckoId: asset.coingecko_id,
+        };
+      });
+      
+
+      const feeCurrencies = chainInfoJson.fees.fee_tokens.map((feeToken: FeeToken) => {
+        // Find the corresponding asset for the fee token
+        const correspondingAsset = assetListJson.assets.find((asset: Asset) => asset.base === feeToken.denom);
+        if (!correspondingAsset) {
+          throw new Error(`No corresponding asset found for fee token denom: ${feeToken.denom}`);
```

Comment:
If you throw an error like this, the app/script using this function is likely to crash if the error is not caught properly. Maybe can revise this implementation such that if 
`!correspondingAsset`, the function skips this token.

Reply:
@sarah-thong In this case would it be fine to use existing information instead?

```
return IBCUtils.EmbedChainInfos[chainId];
```

Reply:
This is not related to the issue. When you throw an error, the whole function fails with an error (i.e. you will not receive any output at all), which is a little stupid if it's only 1 or 2 asset infos that are missing.

As such, it's better to just continue the loop without adding the asset, thereby ensuring you have an output (however incomplete).

Reply:
Thank you @sarah-thong. I have made the fix by filtering out fee tokens which do not have a corresponding asset. This way we will not run into the scenario of having undefined `correspondingAsset`